volatility3/framework/layers/arm.py

```diff
@@ -0,0 +1,475 @@
+# This file is Copyright 2024 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+import functools
+import collections
+from typing import Optional, Dict, Any, List, Iterable, Tuple
+
+from volatility3 import classproperty
+from volatility3.framework import interfaces, exceptions
+from volatility3.framework.configuration import requirements
+from volatility3.framework.layers import linear
+
+vollog = logging.getLogger(__name__)
+AARCH64_TRANSLATION_DEBUGGING = False
+AARCH64_DEBUGGING = False
+
+"""
+Webography :
+ [1] Arm, "Arm Architecture Reference Manual for A-profile architecture, DDI 0487J.a (ID042523)", https://developer.arm.com/documentation/ddi0487/ja/?lang=en
+ [2] Linux, Linux Kernel source code, v6.7
+
+Glossary :
+ TTB : Translation Table Base
+ TCR : Translation Control Register
+ EL : Exception Level (0:Application,1:Kernel,2:Hypervisor,3:Secure Monitor)
+ Granule : Translation granule (smallest block of memory that can be described)
+ """
+
+
+class AArch64Exception(exceptions.LayerException):
+    pass
+
+
+class AArch64(linear.LinearlyMappedLayer):
+    """Translation Layer for the Arm AArch64 memory mapping."""
+
+    _direct_metadata = collections.ChainMap(
+        {"architecture": "AArch64"},
+        {"mapped": True},
+        interfaces.layers.TranslationLayerInterface._direct_metadata,
+    )
+
+    _bits_per_register = 64
+    _register_size = _bits_per_register // 8
+    # NOTE: _maxphyaddr is as defined in the AArch64 specs *NOT* the maximum physical address
+    # _maxvirtaddr actually depends on the context
+    # We need the full 64 bits masking on addresses to determine their TTB at bit 55
+    _maxphyaddr = 64
+    _maxvirtaddr = _maxphyaddr
+
+    def __init__(
+        self,
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        name: str,
+        metadata: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        super().__init__(
+            context=context, config_path=config_path, name=name, metadata=metadata
+        )
+        self._base_layer = self.config["memory_layer"]
+        # self._swap_layers = []  # TODO
+        self._page_map_offset = self.config["page_map_offset"]
+        self._tcr_el1_tnsz = self.config["tcr_el1_tnsz"]
+        self._page_size = self.config["page_size"]
```

Comment:
- `_page_map_offset` is the DTB (either kernel or specific to a process). For the kernel, it can be calculated thanks to the existing Intel implementation.
- `_tcr_el1_tnsz` : *The size offset of the memory region addressed by TTBR1_EL1. The region size is 2(64-T1SZ) bytes* -> See page 7081 of the doc. It is defined in the kernel config, but we cannot predict it (it is calculated in LinuxStacker).
- `_page_size` : Page size (4, 16 or 64 KB), also defined in the kernel config, but we cannot predict it (it is calculated in LinuxStacker).